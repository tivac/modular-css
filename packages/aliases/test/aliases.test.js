"use strict";

const { dedent } = require("dentist");
    
const Processor = require("@modular-css/processor");
const namer     = require("@modular-css/test-utils/namer.js");

const aliases = require("../aliases.js");

describe("@modular-css/path-aliases", () => {
    it("should return a falsey value if a file isn't found", () => {
        const fn = aliases({
            aliases : {
                specimens : "./packages/aliases/test/specimens",
            },
        });

        expect(fn(".", "specimens/fooga.css")).toBeFalsy();
    });

    it("should return the absolute path if a file is found", () => {
        const fn = aliases({
            aliases : {
                one : "./packages/aliases/test/specimens/one",
            },
        });

        expect(fn(".", "one/one.css")).toBe(require.resolve("./specimens/one/one.css"));
    });

    it("should check multiple aliases for files & return the first match", () => {
        const fn = aliases({
            aliases : {
                one : "./packages/aliases/test/specimens/one",
                two : "./packages/aliases/test/specimens/two",
                sub : "./packages/aliases/test/specimens/one/sub",
            },
        });

        expect(fn(".", "one/one.css")).toBe(require.resolve("./specimens/one/one.css"));
        expect(fn(".", "sub/sub.css")).toBe(require.resolve("./specimens/one/sub/sub.css"));
    });

    it("should be usable as a modular-css resolver", async () => {
        const processor = new Processor({
            namer,
            resolvers : [
                aliases({
                    aliases : {
                        sub : "./packages/aliases/test/specimens/one/sub",
                        two : "./packages/aliases/test/specimens/two",
                    },
                }),
            ],
        });
        
        await processor.string(
            "./packages/paths/test/specimens/one/start.css",
            dedent(`
                @value sub from "sub/sub.css";
                
                .rule {
                    composes: two from "two/two.css";
                }
            `)
        );

        const { compositions } = await processor.output();

        expect(compositions).toMatchSnapshot();
    });

    it("should fall through to the default resolver", async () => {
        const processor = new Processor({
            namer,
            resolvers : [
                aliases({
                    aliases : {
                        two : "./packages/aliases/test/specimens/two",
                    },
                }),
            ],
        });
        
        await processor.string(
            "./packages/paths/test/specimens/one/start.css",
            dedent(`
                @value sub from "./sub/sub.css";
                
                .rule {
                    composes: two from "two/two.css";
                }
            `)
        );

        const { compositions } = await processor.output();
        
        expect(compositions).toMatchSnapshot();
    });
});
