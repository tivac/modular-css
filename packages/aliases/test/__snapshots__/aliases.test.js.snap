// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modular-css/path-aliases should be usable as a modular-css resolver 1`] = `
Object {
  "packages/aliases/test/specimens/one/sub/sub.css": Object {
    "sub": Array [
      "sub",
    ],
  },
  "packages/aliases/test/specimens/two/two.css": Object {
    "two": Array [
      "two",
    ],
  },
  "packages/paths/test/specimens/one/start.css": Object {
    "rule": Array [
      "two",
      "rule",
    ],
  },
}
`;

exports[`@modular-css/path-aliases should fall through to the default resolver 1`] = `
Object {
  "packages/aliases/test/specimens/two/two.css": Object {
    "two": Array [
      "two",
    ],
  },
  "packages/paths/test/specimens/one/start.css": Object {
    "rule": Array [
      "two",
      "rule",
    ],
  },
  "packages/paths/test/specimens/one/sub/sub.css": Object {
    "sub": Array [
      "sub",
    ],
  },
}
`;
